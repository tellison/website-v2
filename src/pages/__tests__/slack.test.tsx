import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe';
import Slack, { Head } from '../slack';

describe('Slack page', () => {
  it('renders correctly', () => {
    const { container } = render(<Slack />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('head renders correctly', () => {
    const { container } = render(<Head />);
    // eslint-disable-next-line
    const title = container.querySelector('title');
    expect(title?.textContent).toEqual('Slack Signup | Adoptium');
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<Slack />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
