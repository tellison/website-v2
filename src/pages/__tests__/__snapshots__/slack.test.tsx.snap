// Vitest Snapshot v1

exports[`Slack page > renders correctly 1`] = `
<main>
  <section
    class="py-5 text-center container"
  >
    <div
      class="row py-lg-5"
    >
      <div
        class="col-lg-8 col-md-8 mx-auto"
      >
        <h1
          class="fw-light"
        >
          Join our Slack channel!
        </h1>
      </div>
      <form
        action="https://docs.google.com/forms/u/1/d/e/1FAIpQLSdxrFvVu49Nx0zQKZAfHe_43H3H6BOhRiRXayQjjAmlbnSJfg/formResponse"
        method="post"
        target="_blank"
      >
        <div
          class="form-group"
        >
          <label>
            Email address
            <input
              autocomplete="email"
              class="form-control"
              name="emailAddress"
              placeholder="Your email"
              required=""
              type="email"
            />
          </label>
          <br />
          <button
            class="btn btn-primary mt-3"
            type="submit"
          >
            Invite Me!
          </button>
        </div>
      </form>
    </div>
  </section>
</main>
`;
